ray-operator/controllers/ray/utils/dashboard_httpclient.go

```diff
@@ -352,7 +367,7 @@ func (r *RayDashboardClient) GetJobInfo(ctx context.Context, jobId string) (*Ray
 	defer resp.Body.Close()
 
 	if resp.StatusCode == http.StatusNotFound {
-		return nil, nil
+		return nil, errors.NewBadRequest("Job " + jobId + " does not exist on the cluster")
```

Comment:
That change breaks the RayJob controller, and the Ray cluster dashboard check that's performed prior to submitting the job, and that's been added as part of #1512, with a work-around for #1381.

I think the easier option in the short term is to update the call from the RayJob controller to check the error (better be NotFound error).

@kevin85421 @architkulkarni what do you think, you guys probably have more context on what's the best strategy to handle this?

Reply:
Thank you @astefanutti. For the time being, I rolled back the changes (preserving them as comments). Lets merge this one and move the issue of `the RayJob controller to check the error` to a separate PR

Reply:
Yes, rolling back sounds good so we can merge this PR.  There is another PR in flight which is dealing with this issue: https://github.com/ray-project/kuberay/pull/1623